build/targets/KoreBuild.SolutionBuild.targets

```diff
@@ -41,27 +45,60 @@ as they are subject to change.
 -->
 
   <Target Name="CleanArtifacts">
-    <!-- intentionally BuildDir, not ArtifactDir -->
-    <RemoveDir Directories="$(BuildDir)" />
+    <ItemGroup>
+      <_FilesToDelete Include="$(ArtifactsDir)**\*" Exclude="$(ArtifactsDir)msbuild\**\*" />
+    </ItemGroup>
+    <Delete Files="@(_FilesToDelete)" />
+  </Target>
+
+  <Target Name="_PrepareOutputPaths">
+    <MakeDir Directories="$(ArtifactsDir);$(BuildDir)" />
+  </Target>
+
+  <Target Name="_ResolveCommitHash" Condition="'$(CommitHash)'==''">
+    <PropertyGroup>
+      <CommitHash Condition="'$(APPVEYOR_REPO_COMMIT)' != ''">$(APPVEYOR_REPO_COMMIT)</CommitHash>
+      <CommitHash Condition="'$(TRAVIS_COMMIT)' != ''">$(TRAVIS_COMMIT)</CommitHash>
+    </PropertyGroup>
+
+    <GetGitCommitInfo WorkingDirectory="$(RepositoryRoot)"
+                      Condition="'$(CommitHash)' == ''"
+                      ContinueOnError="WarnAndContinue">
+      <Output TaskParameter="CommitHash" PropertyName="CommitHash" />
+      <Output TaskParameter="Branch" PropertyName="GitBranch" />
+    </GetGitCommitInfo>
+  </Target>
+
+  <!-- The filename that will contain the commit hash -->
+  <PropertyGroup>
+    <_CommitArtifactFileName>commit</_CommitArtifactFileName>
+  </PropertyGroup>
+
+  <Target Name="_CreateCommitHashArtifact" DependsOnTargets="_PrepareOutputPaths;_ResolveCommitHash">
+    <WriteLinesToFile
+      File="$(ArtifactsDir)$(_CommitArtifactFileName)"
+      Overwrite="true"
+      Lines="$(CommitHash)"
+      Condition="'$(CommitHash)' != ''" />
   </Target>
 
 <!--
 ###################################################################
-Targets: UseVolatileFeed
+Targets: _UseVolatileFeed
 
 Replaces $(RepositoryRoot)\NuGet.config feeds.
 ###################################################################
 -->
   <ItemGroup>
-    <UpdateFeeds Include="ARTIFACTS" Value="$(NUGET_VOLATILE_FEED_ARTIFACTS)" Condition="'$(NUGET_VOLATILE_FEED_ARTIFACTS)' != ''" />
-    <UpdateFeeds Include="AspNetCore" Value="$(NUGET_VOLATILE_FEED_AspNetCore)" Condition="'$(NUGET_VOLATILE_FEED_AspNetCore)' != ''" />
+    <_UpdateFeeds Include="ARTIFACTS" Value="$(NUGET_VOLATILE_FEED_ARTIFACTS)" Condition="'$(NUGET_VOLATILE_FEED_ARTIFACTS)' != ''" />
```

Comment:
There is no update feeds target. This is input for the `_UseVolatileFeed` target.